packages/compiler/test/checker/deprecation.test.ts

```diff
@@ -0,0 +1,311 @@
+import {
+  BasicTestRunner,
+  createTestHost,
+  createTestRunner,
+  createTestWrapper,
+  DiagnosticMatch,
+  expectDiagnosticEmpty,
+  expectDiagnostics,
+  extractCursor,
+  TestHost,
+} from "../../src/testing/index.js";
+
+describe("compiler: checker: deprecation", () => {
+  let runner: BasicTestRunner;
+
+  beforeEach(async () => {
+    runner = await createTestRunner();
+  });
+
+  async function expectDeprecations(
+    source: string,
+    deprecations: string[],
+    testRunner: BasicTestRunner = runner
+  ) {
+    const expectedDiagnostics: DiagnosticMatch[] = [];
+    for (const deprecation of deprecations) {
+      const { source: newSource, pos } = extractCursor(source);
+      expectedDiagnostics.push({
+        code: "deprecated",
+        message: `Deprecated: ${deprecation}`,
+        severity: "warning",
+        pos,
+      });
+
+      // Continue with the updated source
+      source = newSource;
+    }
+
+    expectDiagnostics(await testRunner.diagnose(source), expectedDiagnostics);
+  }
+
+  describe("#deprecated directive", () => {
```

Comment:
Should add a test for #suppressing #deprecated?

Reply:
Added a test and found an issue while doing so!  I've now added some logic to skip raising diagnostics when deprecated types are resolved for a template parameter inside of a templated type.